src/objective-c/tests/GRPCClientTests.m

```diff
@@ -39,7 +39,7 @@
 #import <GRPCClient/GRPCCall+OAuth2.h>
 #import <GRPCClient/GRPCCall+Tests.h>
 #import <ProtoRPC/ProtoMethod.h>
-#import <RemoteTest/Messages.pbobjc.h>
+#import <RemoteTest/src/proto/grpc/testing/Messages.pbobjc.h>
```

Comment:
This would prevent us from having Bazel tests to ensure things aren't broken on import. I have a change ready for that, but I'm not sure how it could be made compatible with this change.

Maybe we can split this PR in two: the change in `objective_c_generator.cc`, which is high priority to fix #6834; and the usage of protos under `/src/proto`, for which we can meet and design how to go about it and Bazel testing.